### PR TITLE
Wd 537 create bluetooth ingress module

### PR DESCRIPTION
**Pull request by GhasBarbouchi**

### CODE REVIEW

File: image/src/main.py [hunk 1]
```diff
@@ -11,12 +11,13 @@ def main():
     """ Main app entry point"""
     configure_logging()
 
-    created = create_app()
+    create_app()
+    """
```
Comment (europaul) [inline image/src/main.py]: please remove lines instead of commenting out

File: image/src/app/module/main.py [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+        if len(devices) < 1:
+            raise OSError(
+                'No nearby Devices found. Make sure your Bluetooth Connection '
+                'is on.')
+        else:
+            for dev in devices:
+                #The manufacturer data is advertized data in ble like deice name should be maximum 50 digits if the device name composed by 2 or less digits
+                #Digits could be the hex ocnversion of any or many of [0,1,2,3,4,5,6,7,8,9,1,b,c,d,e.f] ,the letters in the last list could be upper case also
+                if dev.addr==APPLICATION['MAC_ADDR']:
+                        print("device name ", dev.getValueText(9))
```
Comment (europaul) [inline image/src/app/module/main.py]: use named constants instead of magic numbers

File: image/src/app/module/main.py [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
+                        manufData=dev.getValueText(255)
+                        bluetooth_data={"bleData":""}
+                        bluetooth_data['bleData']=str(manufData)
```
Comment (europaul) [inline image/src/app/module/main.py]:
```suggestion
                        bluetooth_data['bleData'] = str(manufData)
```
here and below

File: image/src/app/module/main.py [hunk 4]
```diff
@@ -1,19 +1,41 @@
...
+                'No nearby Devices found. Make sure your Bluetooth Connection '
+                'is on.')
+        else:
+            for dev in devices:
+                #The manufacturer data is advertized data in ble like deice name should be maximum 50 digits if the device name composed by 2 or less digits
+                #Digits could be the hex ocnversion of any or many of [0,1,2,3,4,5,6,7,8,9,1,b,c,d,e.f] ,the letters in the last list could be upper case also
+                if dev.addr==APPLICATION['MAC_ADDR']:
+                        print("device name ", dev.getValueText(9))
+                        manufData=dev.getValueText(255)
+                        bluetooth_data={"bleData":""}
```
Comment (europaul) [inline image/src/app/module/main.py]: bluetooth_data = {
                            'bleData': str(manufData)
}

File: image/src/app/module/main.py [hunk 5]
```diff
@@ -1,19 +1,41 @@
+
+from app.weeve.egress import send_data
+from logging import getLogger
+from app.config import APPLICATION
+import time
```
Comment (europaul) [inline image/src/app/module/main.py]:
```suggestion
from time import sleep
```

File: image/src/app/config/weeve.py [hunk 6]
```diff
@@ -4,11 +4,7 @@
...
-    "MODULE_NAME": env("MODULE_NAME", "python-ingress-boilerplate"),
-    "MODULE_TYPE": env("MODULE_TYPE", "INGRESS"),
-    "EGRESS_SCHEME": env("EGRESS_SCHEME", "http"),
-    "EGRESS_HOST": env("EGRESS_HOST", "0.0.0.0"),
-    "EGRESS_PORT": env("EGRESS_PORT", "80"),
-    "EGRESS_PATH": env("EGRESS_PATH", ""),
-    "EGRESS_URL": env("EGRESS_URL", "http://localhost:8000")
+"MODULE_NAME": env("MODULE_NAME", "bluetooth-ingress"),
+"MODULE_TYPE": env("MODULE_TYPE", "INGRESS"),
+"EGRESS_URL": env("EGRESS_URL", "http://172.17.0.2:80")
```
Comment (europaul) [inline image/src/app/config/weeve.py]: please revert

File: image/src/app/config/weeve.py [hunk 7]
```diff
@@ -4,11 +4,7 @@
...
 
 WEEVE = {
-    "MODULE_NAME": env("MODULE_NAME", "python-ingress-boilerplate"),
-    "MODULE_TYPE": env("MODULE_TYPE", "INGRESS"),
-    "EGRESS_SCHEME": env("EGRESS_SCHEME", "http"),
-    "EGRESS_HOST": env("EGRESS_HOST", "0.0.0.0"),
-    "EGRESS_PORT": env("EGRESS_PORT", "80"),
-    "EGRESS_PATH": env("EGRESS_PATH", ""),
-    "EGRESS_URL": env("EGRESS_URL", "http://localhost:8000")
+"MODULE_NAME": env("MODULE_NAME", "bluetooth-ingress"),
```
Comment (europaul) [inline image/src/app/config/weeve.py]: please revert the unindent

File: image/Dockerfile [hunk 8]
```diff
@@ -2,7 +2,9 @@ FROM python:3.8-slim
 WORKDIR /app
 RUN mkdir -p /app
 COPY ./image/requirements.txt /app
-RUN pip install -r requirements.txt
+RUN   apt-get update -y && apt-get upgrade -y && apt-get install -y make
```
Comment (europaul) [inline image/Dockerfile]: remove `apt-get upgrade`

Comment (europaul) [reply image/Dockerfile]: remove `-y` from update

Comment (europaul) [reply image/Dockerfile]: put `install make` in the next line

File: image/Dockerfile [hunk 9]
```diff
@@ -2,7 +2,9 @@ FROM python:3.8-slim
...
+RUN apt-get install  libglib2.0-dev python3-pip  python3-dev -y
```
Comment (europaul) [inline image/Dockerfile]: remove unnecessary packets

File: docker-compose.yml [hunk 10]
```diff
@@ -5,10 +5,11 @@ services:
       context: .
       dockerfile: Dockerfile
     environment:
-      MODULE_NAME: python-ingress-boilerplate
+      MODULE_NAME: bluetooth-ingress
       EGRESS_URL: https://hookb.in/r1YwjDyn7BHzWWJVK8Gq
       HANDLER_PORT: 80
```
Comment (europaul) [inline docker-compose.yml]: please remove

File: config.env [hunk 11]
```diff
@@ -1,7 +1,10 @@
 # GENERAL CONFIGURATION VARIABLES
 
-MODULE_NAME=python-ingress-boilerplate
+MODULE_NAME=bluetooth-ingress
 MODULE_TYPE=INGRESS
-EGRESS_URL=https://hookb.in/3OYoaZXXV9c7yakkyyM6
+EGRESS_URL= http://172.17.0.2:80
+SCAN_TIMEOUT=3
```
Comment (europaul) [inline config.env:6]: please be consistent

File: config.env [hunk 12]
```diff
@@ -1,7 +1,10 @@
 # GENERAL CONFIGURATION VARIABLES
 
-MODULE_NAME=python-ingress-boilerplate
+MODULE_NAME=bluetooth-ingress
 MODULE_TYPE=INGRESS
-EGRESS_URL=https://hookb.in/3OYoaZXXV9c7yakkyyM6
+EGRESS_URL= http://172.17.0.2:80
```
Comment (europaul) [inline config.env]: please revert

File: Module.yaml [hunk 13]
```diff
@@ -13,21 +13,27 @@ categories:
...
+    default: '00:10:18:01:4b:b5'
     type: string
-    name: Output Unit
+    name: Mac Address
+  - key: PERIOD
+    description: The application scan and read the ble data every choosen period
+    value: ''
+    default: 5
+    type: integer
+    name: Period between two scans tiemout (secs)
```
Comment (europaul) [inline Module.yaml]: this looks more like a description, the name is usually shorter

**Review by europaul — CHANGES_REQUESTED** [review]

File: image/Dockerfile (hunk 9 above)
Comment (GhasBarbouchi) [reply image/Dockerfile]: when I removed the "apt-get install python3-pip" packet the pip install of bluepy returned error and failed to install it.

File: image/src/app/module/main.py [hunk 14]
```diff
@@ -1,19 +1,40 @@
...
+        for dev in devices:
+            #The manufacturer data is advertized data in ble like deice name
+            # should be maximum 50 digits if the device name composed by 2 or less digits
+            #Digits could be the hex ocnversion of any or many of [0,1,2,3,4,5,6,7,8,9,1,b,c,d,e.f]
+            # the letters in the last list could be upper case also
+            if dev.addr == APPLICATION['MAC_ADDR']:
+                print("device name ", dev.getValueText(DEVICE_NAME))
+                manuf_data = dev.getValueText(DEVICE_MANUF_DATA)
+                bluetooth_data = {'bleData': str(manuf_data) }
+                print("ble Dta",bluetooth_data)
```
Comment (europaul) [inline image/src/app/module/main.py]: please fix the typo

File: image/src/app/config/weeve.py [hunk 15]
```diff
@@ -4,11 +4,7 @@
...
-    "MODULE_NAME": env("MODULE_NAME", "python-ingress-boilerplate"),
-    "MODULE_TYPE": env("MODULE_TYPE", "INGRESS"),
-    "EGRESS_SCHEME": env("EGRESS_SCHEME", "http"),
-    "EGRESS_HOST": env("EGRESS_HOST", "0.0.0.0"),
-    "EGRESS_PORT": env("EGRESS_PORT", "80"),
-    "EGRESS_PATH": env("EGRESS_PATH", ""),
-    "EGRESS_URL": env("EGRESS_URL", "http://localhost:8000")
+    "MODULE_NAME" : env("MODULE_NAME", "bluetooth-ingress"),
+    "MODULE_TYPE" : env("MODULE_TYPE", "INGRESS"),
+    "EGRESS_URL"  : env("EGRESS_URL", " https://hookb.in/r1YwjDyn7BHzWWJVK8Gq")
```
Comment (europaul) [inline image/src/app/config/weeve.py]: please remove the space in front of the url

File: image/Dockerfile [hunk 16]
```diff
@@ -2,7 +2,9 @@ FROM python:3.8-slim
 WORKDIR /app
 RUN mkdir -p /app
 COPY ./image/requirements.txt /app
-RUN pip install -r requirements.txt
+RUN apt-get update
+RUN apt-get install make -y libglib2.0-dev -y python3-pip -y
```
Comment (europaul) [inline image/Dockerfile]: I think one `-y` should be enough

Comment (europaul) [reply image/Dockerfile]: also pip is already installed in this image

File: image/Dockerfile (hunk 9 above)
Comment (europaul) [reply image/Dockerfile]: what's the error?

**Review by europaul — CHANGES_REQUESTED** [review]
mostly have to fix typos

File: image/Dockerfile (hunk 9 above)
Comment (GhasBarbouchi) [reply image/Dockerfile]: error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [13 lines of output]
      /usr/local/lib/python3.8/site-packages/setuptools/dist.py:697: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
        warnings.warn(
      running bdist_wheel
      running build
      running build_py
      Working dir is /tmp/pip-install-kx4ft85m/bluepy_d60630f289bb438b894af259dc13c020
      execute make -C ./bluepy clean
      execute make -C bluepy -j1
      Failed to compile bluepy-helper. Exiting install.
      Command was 'make -C bluepy -j1' in /tmp/pip-install-kx4ft85m/bluepy_d60630f289bb438b894af259dc13c020
      Return code was 2
      Output was:
      b"make: Entering directory '/tmp/pip-install-kx4ft85m/bluepy_d60630f289bb438b894af259dc13c020/bluepy'\ntar xzf bluez-src.tgz\ntouch ./bluez-5.47/lib/bluetooth.c ./bluez-5.47/lib/hci.c ./bluez-5.47/lib/sdp.c ./bluez-5.47/lib/uuid.c ./bluez-5.47/attrib/att.c ./bluez-5.47/attrib/gatt.c ./bluez-5.47/attrib/gattrib.c ./bluez-5.47/attrib/utils.c ./bluez-5.47/btio/btio.c ./bluez-5.47/src/log.c ./bluez-5.47/src/shared/mgmt.c ./bluez-5.47/src/shared/crypto.c ./bluez-5.47/src/shared/att.c ./bluez-5.47/src/shared/queue.c ./bluez-5.47/src/shared/util.c ./bluez-5.47/src/shared/io-glib.c ./bluez-5.47/src/shared/timeout-glib.c\ncc -L. -g -Wall  -Os -DHAVE_CONFIG_H -I./bluez-5.47/attrib -I./bluez-5.47 -I./bluez-5.47/lib -I./bluez-5.47/src -I./bluez-5.47/gdbus -I./bluez-5.47/btio -I./bluez-5.47/sys -I/usr/include/glib-2.0 -I/usr/lib/aarch64-linux-gnu/glib-2.0/include -o bluepy-helper bluepy-helper.c ./bluez-5.47/lib/bluetooth.c ./bluez-5.47/lib/hci.c ./bluez-5.47/lib/sdp.c ./bluez-5.47/lib/uuid.c ./bluez-5.47/attrib/att.c ./bluez-5.47/attrib/gatt.c ./bluez-5.47/attrib/gattrib.c ./bluez-5.47/attrib/utils.c ./bluez-5.47/btio/btio.c ./bluez-5.47/src/log.c ./bluez-5.47/src/shared/mgmt.c ./bluez-5.47/src/shared/crypto.c ./bluez-5.47/src/shared/att.c ./bluez-5.47/src/shared/queue.c ./bluez-5.47/src/shared/util.c ./bluez-5.47/src/shared/io-glib.c ./bluez-5.47/src/shared/timeout-glib.c -lglib-2.0\nmake: cc: No such file or directory\nmake: *** [Makefile:30: bluepy-helper] Error 127\nmake: Leaving directory '/tmp/pip-install-kx4ft85m/bluepy_d60630f289bb438b894af259dc13c020/bluepy'\n"
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for bluepy

Comment (GhasBarbouchi) [reply image/Dockerfile]: the error above happened when i removed install python3-pip , when the image trying to install bluepy

File: image/src/app/module/main.py (hunk 14 above)
Comment (GhasBarbouchi) [reply image/src/app/module/main.py]: i also changed the name to bluetooth observer ingress , because I we also will develop connected Bluetooth ingress

Comment (europaul) [reply image/src/app/module/main.py]: ok, I renamed the repo as well

File: image/Dockerfile (hunk 9 above)
Comment (europaul) [reply image/Dockerfile]: please look closely at this error message and try to figure out what the problem is

Comment (GhasBarbouchi) [reply image/Dockerfile]: I still figuring it out, looks bluepy-helper like didn't find 'nmake' and  

This error originates from a subprocess and is likely not a problem with pip.

above the error message said not a problem with pip, could be solved with install stable version of bluepy , if I am wrong to tell me to continue learning about that error

**Review by europaul — APPROVED** [review]

**Pull request merged**